### PR TITLE
Ryanunderhill/ocr custom op

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Refactors the custom op code into a separate file
Fixes a minor bug in the C API sample file (memory leak)
Adds more complex inputs to the shared library test functions

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
          "component":{
             "type":"git",
             "git":{
-               "commitHash":"fdb978a580bcb461ea077b9772c973928b8dbf5f",
```
Comment (snnn) [inline cgmanifest.json]: Please revert this change

Comment (RyanUnderhill) [reply cgmanifest.json]: Good catch, I didn't mean to check that in.
